public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -62,28 +61,27 @@ export function PrometheusDatasource(instanceSettings, $q, backendSrv, templateS
     var start = getPrometheusTime(options.range.from, false);
     var end = getPrometheusTime(options.range.to, true);
 
-    var queries = [];
-    options = _.clone(options);
-    _.each(options.targets, _.bind(function(target) {
+    var queries = _.map(options.targets, target => {
       if (!target.expr || target.hide) {
         return;
       }
 
       var query: any = {};
-      query.expr = templateSrv.replace(target.expr, options.scopedVars, interpolateQueryExpr);
+      var interval = templateSrv.replace(target.interval || options.interval);
+      query.step = this.calculateStep(interval);
 
-      var interval = target.interval || options.interval;
-      var intervalFactor = target.intervalFactor || 1;
-      target.step = query.step = this.calculateInterval(interval, intervalFactor);
       var range = Math.ceil(end - start);
       // Prometheus drop query if range/step > 11000
       // calibrate step if it is too big
-      if (query.step !== 0 && range / query.step > 11000) {
-        target.step = query.step = Math.ceil(range / 11000);
+      if (range / query.step > 11000) {
+        query.step = Math.ceil(range / 11000);
```

Comment:
@torkelo Please let me do the supplementary explanation of this comment.
I think, the plugin should re-calculate "interval" from "step" at the line.

Before this line, the "step" is calculated from "interval", these values are syncrhonized. (If "interval" is "10s", the "step" is "10", right?)
But, after this line, these values could be not synchronized. ("interval" is "10s", but the "step" become like "50")

Refer comment from original PR, https://github.com/grafana/grafana/pull/3757#issuecomment-191713400,
> Now say the step is auto-calcualted as 1h (on longer periods).
Every 1h datapoint is calculated as avg from values of now() and now()-5m. It's not correct, it should be of now() and now()-1h.

He want to sync the "interval" and "step" after the "step" is auto-calculated.

So, I think, the plugin should re-calculate synced "interval" from "step".